tests/commands/test_deploy.py

```diff
@@ -22,78 +22,54 @@ def tmp_working_dir(monkeypatch, tmp_path):
 ABI_OVERRIDE = f"{ABIS_DIRECTORY}/override.json"
 BASE_PATH = (BUILD_DIRECTORY, ABIS_DIRECTORY)
 PATH_OVERRIDE = ("artifacts2", ABIS_DIRECTORY)
-RUN_OUTPUT = b"output"
 ARGS = [1, 2, 3]
 ADDRESS = 999
 TX_HASH = 222
+CALL_OUTPUT = [ADDRESS, TX_HASH]
 
 
+@pytest.mark.asyncio
 @pytest.mark.parametrize(
-    "args, exp_command, exp_abi",
+    "args, exp_abi",
     [
         (
             [CONTRACT, ARGS, NETWORK, ALIAS],  # args
-            {  # expected command
-                "contract_name": CONTRACT,
-                "network": NETWORK,
-                "overriding_path": None,
-                "mainnet_token": None,
-            },
             ABI,  # expected ABI
         ),
         (
             [CONTRACT, ARGS, NETWORK, ALIAS, PATH_OVERRIDE],  # args
-            {  # expected command
-                "contract_name": CONTRACT,
-                "network": NETWORK,
-                "overriding_path": PATH_OVERRIDE,
-                "mainnet_token": None,
-            },
             ABI,  # expected ABI
         ),
         (
             [CONTRACT, ARGS, NETWORK, ALIAS, None, ABI_OVERRIDE],  # args
-            {  # expected command
-                "contract_name": CONTRACT,
-                "network": NETWORK,
-                "overriding_path": None,
-                "mainnet_token": None,
-            },
             ABI_OVERRIDE,  # expected ABI
         ),
         (
             [CONTRACT, ARGS, NETWORK, ALIAS, PATH_OVERRIDE, ABI_OVERRIDE],  # args
-            {  # expected command
-                "contract_name": CONTRACT,
-                "network": NETWORK,
-                "overriding_path": PATH_OVERRIDE,
-                "mainnet_token": None,
-            },
             ABI_OVERRIDE,  # expected ABI
         ),
     ],
 )
-@patch("nile.core.deploy.run_command", return_value=RUN_OUTPUT)
-@patch("nile.core.deploy.parse_information", return_value=[ADDRESS, TX_HASH])
+@patch("nile.core.deploy.parse_information", return_value=CALL_OUTPUT)
 @patch("nile.core.deploy.deployments.register")
-def test_deploy(
-    mock_register, mock_parse, mock_run_cmd, caplog, args, exp_command, exp_abi
-):
+async def test_deploy(mock_register, mock_parse, caplog, args, exp_abi):
     logging.getLogger().setLevel(logging.INFO)
 
-    # check return values
-    res = deploy(*args)
-    assert res == (ADDRESS, exp_abi)
+    with patch("nile.core.deploy.call_cli", new=AsyncMock()) as mock_cli_call:
+        mock_cli_call.return_value = CALL_OUTPUT
```

Comment:
shouldn't we test that `call_cli` (or starknet_cli) is called with the right parameters? why removing the equivalent of `mock_run_cmd.assert_called_once_with`?

Reply:
Yes, we should. I must have deleted this by mistake during one of the conflict fixes